python/python/cwrap/stream.py

```diff
@@ -0,0 +1,177 @@
+#  Copyright (C) 2016  Statoil ASA, Norway.
+#
+#  This file is part of ERT - Ensemble based Reservoir Tool.
+#
+#  ERT is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+#  FITNESS FOR A PARTICULAR PURPOSE.
+#
+#  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+#  for more details.
+
+from os import linesep
+import numpy
+from .clib import load as cwrapload
+from .prototype import Prototype
+from .basecclass import BaseCClass
+
+from six import string_types
+
+class LibcPrototype(Prototype):
+    lib = cwrapload('libc', '.6')
+
+    def __init__(self, prototype, bind=False):
+        super(LibcPrototype, self).__init__(LibcPrototype.lib, prototype, bind=bind)
+
+class Stream(BaseCClass):
+    """
+    Utility class to map a Python file handle <-> FILE* in C
+    """
+    TYPE_NAME = "stream"
+
+    _fopen  = LibcPrototype("void* fopen(char*, char*)")
+    _fread  = LibcPrototype("int   fread(void*, int, int, stream)")
+    _fwrite = LibcPrototype("int   fwrite(void*, int, int, stream)")
+    _fseek  = LibcPrototype("int   fseek(stream, int, int)", bind = True)
+
+    _fclose = LibcPrototype("int   fclose (stream)", bind = True);
+    _fflush = LibcPrototype("int   fflush (stream)", bind = True);
+
+    def __init__(self, fname, mode = 'rb'):
+        c_ptr = self._fopen(fname, mode)
+        self._mode = mode
+        self._fname = fname
+
+        self.closed = False # A closed file cannot be used for further I/O
```

Comment:
Shouldn't this be prefixed with `_`and then have a read only property for closed?